File: VB/DxBlazorApplication1/Services/WeatherForecast.vb
Imports System.ComponentModel.DataAnnotations

Namespace DxBlazorApplication1.Services

    Public Class WeatherForecast

        <Key>
        Public Property ID As Integer

        <Required>
        Public Property [Date] As DateTime

        <Range(0, 100)>
        Public Property TemperatureC As Integer

        Public ReadOnly Property TemperatureF As Integer
            Get
                Return 32 + CInt(TemperatureC / 0.5556)
            End Get
        End Property

        <Required>
        <MaxLength(50)>
        Public Property Summary As String?
    End Class
End Namespace

File: CS/BlazorGridPopupEditFormCustomization/Services/WeatherForecast.cs
using System.ComponentModel.DataAnnotations;

namespace BlazorGridPopupEditFormCustomization.Services {
    public class WeatherForecast {
        [Key]
        public int ID { get; set; }
        [Required]
        public DateTime Date { get; set; }
        [Range(0, 100)]
        public int TemperatureC { get; set; }

        public int TemperatureF => 32 + (int)(TemperatureC / 0.5556);
        [Required]
        [MaxLength(50)]
        public string? Summary { get; set; }
    }
}